Feature/tsq9ogs8 8 request list page

an-nguyen-jh opened the pull request.

an-nguyen-jh committed:
[Feat] Navigate between item list page and request list page

an-nguyen-jh committed:
[Style] Create table view for device requests

Create table view for 3 types of request: resolve, deny, pending

Add deny and resolve button

an-nguyen-jh committed:
[Style] Switch between 3 type of table

an-nguyen-jh committed:
[Feat] Load all device request separate by type

an-nguyen-jh committed:
[Feat] Navigate between page in each table

an-nguyen-jh committed:
[Fix] Fix Pagination re render to many time

an-nguyen-jh committed:
[Feat] Handle deny employee's device request

an-nguyen-jh committed:
[Feat] Handle resolve request function anf update device info

an-nguyen-jh committed:
[Fix] fix mistake in css file and remove unnecessary

anh-dinh-jh commented on src/components/DeviceRequest/DeviceRequest.js:
```diff
@@ -1,11 +1,207 @@
+];
 
 function DeviceRequest() {
+  const [reload, setReload] = useState(Math.random());
```
What is the purpose of this Math.random?

an-nguyen-jh replied:
It isn't necessary.

anh-dinh-jh commented on src/utils/sort.js:
```diff
@@ -17,6 +17,14 @@ const sortPrototypes = {
       return b.updatedTime > a.updatedTime ? 1 : -1;
     },
   },
+  createdTime: {
+    asc: (a, b) => {
+      return a.createdTime > b.createdTime ? 1 : -1;
+    },
+    desc: (a, b) => {
+      return b.createdTime > a.createdTime ? 1 : -1;
+    },
+  },
```
Can we combine this one with `updatedTime` sort?

an-nguyen-jh replied:
I can not find possible ways to do this.

an-nguyen-jh committed:
[Fix] Remove Math.random in the initial value of state

Change pageLimit in Table

an-nguyen-jh committed:
[Fix] Change reload logic of multiple page

Remove Math.random() in Item Details page, Items List page, device request list page

Increase reload number every time something change
